src/authentication/index.js

```diff
@@ -88,8 +93,9 @@ function Authentication(auth0, options) {
     this.baseOptions._sendTelemetry === false
       ? this.baseOptions._sendTelemetry
       : true;
-
-  this.baseOptions.rootUrl = 'https://' + this.baseOptions.domain;
+  
+  this.baseOptions.domainScheme = !this.baseOptions.domainScheme ? 'https://' : this.baseOptions.domainScheme
+  this.baseOptions.rootUrl = this.baseOptions.domainScheme + this.baseOptions.domain;
```

Comment:
What if we would just allow for the domain to include http(s)?

Something along the lines of:

```
this.baseOptions.rootUrl = this.baseOptions.domain.indexOf('http') === 0
  ? this.baseOptions.domain
  : 'https://' + this.baseOptions.domain;
```

This avoids the need for the new option, `domainScheme`, while keeping backwards compatibility.

Reply:
Agreed, I think auto appending a default scheme if either `http` or `https` was not found would be preferable to having an additional option, unless @danmastrowcoles there is a use case we're not considering here.

Reply:
Great suggestions, I've refactored to no longer use the additonal option now